lib/recurly/pricing/checkout/index.js

```diff
@@ -210,30 +246,69 @@ export default class CheckoutPricing extends Pricing {
         }
       }
 
-      let adjustment = {};
-
-      if (existingAdjustment) {
-        // Only pick present attributes
-        if (typeof amount !== 'undefined') adjustment.amount = amount;
-        if (typeof currency !== 'undefined') adjustment.currency = currency;
-        if (typeof quantity !== 'undefined') adjustment.quantity = parseInt(quantity, 10);
-        if (typeof taxExempt !== 'undefined') adjustment.taxExempt = taxExempt;
-        if (typeof taxCode !== 'undefined') adjustment.taxCode = taxCode;
-
-        Object.assign(existingAdjustment, adjustment);
-      } else {
-        // Adjustment defaults
-        quantity = parseInt(quantity, 10);
-        if (isNaN(quantity)) quantity = 1;
-        currency = currency || this.items.currency;
-        taxExempt = !!taxExempt;
+      if (existingAdjustment && existingAdjustment.itemCode && !itemCode) {
+        itemCode = existingAdjustment.itemCode;
+      }
 
-        adjustment = { amount, quantity, id, currency, taxExempt, taxCode };
+      new Promise((resolve, reject) => {
+        if (!itemCode) return resolve();
+        return recurly.item({ itemCode }).then(item => {
+          const expectedCurrency = currency
+                                || (existingAdjustment && existingAdjustment.currency)
+                                || this.items.currency;
```

Comment:
Establishing the expected currency has three possible cascading values.

1. pass the currency explicitly
```js
checkoutPricing.adjustment({ itemcode: 'my-item-code', currency: 'USD' });
```
2. If updating an adjustment, defer to the currency it is currently set to
```js
checkoutPricing.adjustment({ id: 'my-adj', itemCode: 'my-item-code', currency: 'CAD' });
checkoutPricing.adjustment({ id: 'my-adj', amount: 20 }); // currency: 'CAD'
```
3. In all other cases, defer to the default currecy on the `CheckoutPricing` isntance
```js
checkoutPricing
  .currency('EUR')
  .adjustment({ id: 'my-adj', itemCode: 'my-item-code' }); // currency: 'EUR'
```

Reply:
That prioritization makes perfect sense. Is this worth documenting somewhere? It's already logical and what I would expect, so might be unnecessary.

Reply:
I'll add a note to the public documentation once we transition it from `SubscriptionPricing` to `CheckoutPricing`